jsx-runtime/src/index.js

```diff
@@ -27,17 +27,19 @@ const isArray = Array.isArray;
  * @param {unknown} [__self]
  */
 function createVNode(type, props, key, isStaticChildren, __source, __self) {
+	if (!props) props = {};
 	// We'll want to preserve `ref` in props to get rid of the need for
 	// forwardRef components in the future, but that should happen via
 	// a separate PR.
-	let normalizedProps = {},
+	let normalizedProps = props,
 		ref,
 		i;
-	for (i in props) {
-		if (i == 'ref') {
-			ref = props[i];
-		} else {
-			normalizedProps[i] = props[i];
+	if ((ref = normalizedProps.ref)) {
+		normalizedProps = {};
+		for (i in props) {
+			if (i != 'ref') {
+				normalizedProps[i] = props[i];
+			}
 		}
 	}
```

Comment:
I think this might be slightly faster:

```suggestion
	if ('ref' in normalizedProps) {
		normalizedProps = {};
		for (i in props) {
			if (i != 'ref') {
				ref = props[i];
			} else {
				normalizedProps[i] = props[i];
			}
		}
	}
```